src/renderer/dx/DxRenderer.cpp

```diff
@@ -1978,15 +1978,30 @@ CATCH_RETURN()
 
 // Routine Description:
 // - Updates the font used for drawing
+// - This is the version that complies with the IRenderEngine interface
 // Arguments:
 // - pfiFontInfoDesired - Information specifying the font that is requested
 // - fiFontInfo - Filled with the nearest font actually chosen for drawing
 // Return Value:
 // - S_OK or relevant DirectX error
 [[nodiscard]] HRESULT DxEngine::UpdateFont(const FontInfoDesired& pfiFontInfoDesired, FontInfo& fiFontInfo) noexcept
+{
+    return UpdateFont(pfiFontInfoDesired, fiFontInfo, {}, {});
+}
+
+// Routine Description:
+// - Updates the font used for drawing
+// Arguments:
+// - pfiFontInfoDesired - Information specifying the font that is requested
+// - fiFontInfo - Filled with the nearest font actually chosen for drawing
+// - features - The map of font features to use
+// - axes - The map of font axes to use
+// Return Value:
+// - S_OK or relevant DirectX error
+[[nodiscard]] HRESULT DxEngine::UpdateFont(const FontInfoDesired& pfiFontInfoDesired, FontInfo& fiFontInfo, std::unordered_map<std::wstring_view, uint32_t> features, std::unordered_map<std::wstring_view, int32_t> axes) noexcept

```

Comment:
careful with this method signature! It will copy the maps... which I don't think that you want here? Shouldn't this be `const&` up until the point at which they need to be copied so that they can be edited?

(looking at DxFontRenderData, it never edits the features or axes maps in-place...)

Reply:
Yep, the maps are now `const&` all the way